src/main/java/seedu/apollo/command/task/DateCommand.java

```diff
@@ -24,6 +24,8 @@ public class DateCommand extends Command {
      * @throws InvalidDateTime If the input date does not fit the above format.
      */
     public DateCommand (String dateString) throws InvalidDateTime {
+        assert dateString != null : "Date string should not be null";
+        assert dateString.matches("\\d{4}-\\d{2}-\\d{2}") : "Date string should be in format yyyy-MM-dd";
```

Comment:
Formatting for dateString has not been implemented in Deadline and Event yet, can add this line back later.